ni_measurementlink_service/_internal/discovery_client.py

```diff
@@ -254,19 +254,27 @@ def _key_file_exists(key_file_path: pathlib.Path) -> bool:
 
 def _start_service(exe_file_path: pathlib.PurePath, key_file_path: pathlib.Path) -> None:
     """Starts the service at the specified path and wait for the service to get up and running."""
-    subprocess.Popen([exe_file_path], cwd=exe_file_path.parent)
-    # After the execution of process, check for key file existence in the path
-    # stop checking after 30 seconds have elapsed and throw error
-    timeout_time = time.time() + _START_SERVICE_TIMEOUT
-    while True:
-        try:
-            with _open_key_file(str(key_file_path)) as _:
-                return
-        except IOError:
-            pass
-        if time.time() >= timeout_time:
-            raise TimeoutError("Timed out waiting for discovery service to start")
-        time.sleep(_START_SERVICE_POLLING_INTERVAL)
+    try:
+        subprocess.Popen(
+            [exe_file_path],
+            cwd=exe_file_path.parent,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+        # After the execution of process, check for key file existence in the path
+        # stop checking after 30 seconds have elapsed and throw error
+        timeout_time = time.time() + _START_SERVICE_TIMEOUT
+        while True:
+            try:
+                with _open_key_file(str(key_file_path)) as _:
+                    return
+            except IOError:
+                pass
+            if time.time() >= timeout_time:
+                raise TimeoutError("Timed out waiting for discovery service to start")
+            time.sleep(_START_SERVICE_POLLING_INTERVAL)
+    except FileNotFoundError:
```

Comment:
why is the try/except needed?

Reply:
https://github.com/ni/measurementlink-python/blob/main/ni_measurementlink_service/_internal/discovery_client.py#L87

In the above function, exception is handled in such a way that when `FileNotFoundError` happens, `True` is returned which denotes that measurement service is registered successfully.

![image](https://github.com/ni/measurementlink-python/assets/13461073/210cf1b6-8bf1-45f9-b5bd-116ce6094fc9)

But the real scenario is that, when there is `FileNotFoundError` happens from `subprocess.popen`, measurement service is not registered successfully.

Reply:
You cannot solve every problem by writing more code. Please think about why the `register_measurement_service` method catches `FileNotFoundError` in the first place and update the existing code.

Also, replacing a descriptive exception with an empty `Exception` seems like a bad practice.

Reply:
Sure Brad. I have updated the code to handle exception in `register_measurement_service`